src/EditorFeatures/Core/Shared/Extensions/IRefactorNotifyServiceExtensions.cs

```diff
@@ -2,12 +2,48 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System;
 using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.ErrorReporting;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Rename;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.Editor.Shared.Extensions
 {
     internal static class IRefactorNotifyServiceExtensions
     {
+        /// <summary>
+        /// Calls all IRefactorNotifyService implementations TryOnBeforeGlobalSymbolRenamed, and if it succeds calls
+        /// TryOnAfterGlobalSymbolRenamed. All calls are made on the UI thread, the <see cref="ForegroundThreadAffinitizedObject"/>
+        /// is used to ensure this behavior. 
+        /// </summary>
+        public static void TryNotifyChangesSynchronously(
+            this IEnumerable<Lazy<IRefactorNotifyService>> refactorNotifyServices,
+            Workspace workspace,
+            Solution newSolution,
+            Solution oldSolution,
+            IThreadingContext threadContext,
+            CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                var refactorNotifyTask = refactorNotifyServices.TryNotifyChangesAsync(workspace, newSolution, oldSolution, threadContext, cancellationToken);
+                refactorNotifyTask.Wait(cancellationToken);
```

Comment:
@sharwell ?  i feel like we should be using JTF here if we absolutely must be synchronous.